R/stats-prcomp-tidiers.R

```diff
@@ -108,38 +108,17 @@ tidy.prcomp <- function(x, matrix = "u", ...) {
       new_column = "PC"
     )
   } else if (matrix %in% c("rotation", "variables", "v", "loadings")) {
-    labels <- if (is.null(rownames(x$rotation))) {
-      1:nrow(x$rotation)
-    } else {
-      rownames(x$rotation)
-    }
-    variables <- tidyr::pivot_longer(as.data.frame(x$rotation),
-      cols = dplyr::everything(),
-      names_to = "key",
-      values_to = "value"
-    ) %>%
-      tibble::remove_rownames() %>%
-      as.data.frame()
-    ret <- data.frame(
-      label = rep(labels, times = ncomp),
-      variables,
-      stringsAsFactors = FALSE
-    )
-    names(ret) <- c("column", "PC", "value")
+    ret <- x$rotation %>% 
+      tibble::as_tibble(rownames = "labels") %>% 
+      tidyr::pivot_longer(cols = -"labels", 
```

Comment:
apologies, I've misnamed the output column here, which I think triggered a failed test. I think this would be the correct name:

```suggestion
      tibble::as_tibble(rownames = "column") %>% 
      tidyr::pivot_longer(cols = -"column", 
```